grant-project/zcash-grant-system
# Request for Proposal (Pt. 2 - Public View)

Another piece of #84, also closes #123. Relies on #120.

### What This Does

Second part of RFPs, this adds a public view at `/requests` and `/requests/:id`. The views still can't have proposals created on them, but list active and past RFPs. There's also some changes to the header to accomodate a new link. Full set of changes:

* Add list page `/requests` and detail page `/requests/:id`, and associated backend API endpoints.
* Add "Requests" link to header, change "Browse" to "Proposals". Shrinks links and adjusts breakpoints a bit.
* Refactors `ProposalCard` to use a common `Card` component shared with `RFPCard`.

### Steps to Test

* Go to `/requests`, confirm you get empty placeholder.
* Create an RFP via admin in draft state, refresh `/requests`, confirm it's still empty
* Switch RFP to live, confirm it now shows up on `/requests`
* Click on the card, confirm it takes you to a correctly rendered page
* Switch RFP to closed, confirm a closed section now pops up on `/requests` with the card, and open requests shows a placeholder

## Screenshots

![screen shot 2019-01-30 at 2 34 07 am](https://user-images.githubusercontent.com/649992/51966760-6785c080-243b-11e9-90c4-dc862b52a3c4.png)

<sup>Text is placeholder, for ZCF to decide what they want it to say.</sup>

![screen shot 2019-01-30 at 2 37 05 am](https://user-images.githubusercontent.com/649992/51966764-69e81a80-243b-11e9-984d-20d958e02e1f.png)
